laws/src/main/scala/cats/laws/discipline/MiniFloat.scala

```diff
@@ -0,0 +1,177 @@
+      } else if (exponent > maxExponent) {
+        try {
+          val ordersOfMagnitudeToShift = Math.subtractExact(exponent, maxExponent)
+
+          val proposedSignificand: Int = Math.multiplyExact(
+            significand,
+            math.pow(base.toDouble, ordersOfMagnitudeToShift.toDouble).toInt
+          )
+          val proposedExponent: Int = Math.subtractExact(exponent, ordersOfMagnitudeToShift)
+
+          if (withinBounds(proposedSignificand, proposedExponent)) {
+            Some(new Finite(proposedSignificand, proposedExponent))
+          } else {
+            None
+          }
+        } catch {
+          case _: ArithmeticException => None
+        }
+      } else {
+        None
+      }
+    }
+
+    private def withinBounds(significand: Int, exponent: Int): Boolean =
+      (minExponent <= exponent && exponent <= maxExponent) &&
+        (minSignificand <= significand && significand <= maxSignificand)
+
+    private val floatExponentStartBit: Int = 23
+    private val floatExponentLength: Int = 8
+    private val floatExponentBias: Int = 127
+    private val floatExponentMask: Int = ((1 << floatExponentLength) - 1) << floatExponentStartBit
+
+    // This does the same thing as java.lang.Math.getExponent, but that method is not available in scalaJS so we have to
+    // do the same thing here.
+    private def getExponent(float: Float): Int =
+      ((floatExponentMask & java.lang.Float.floatToIntBits(float)) >> floatExponentStartBit) - floatExponentBias
+
+  }
+
+  val Zero: MiniFloat = Finite.zero
+  val NegativeOne: MiniFloat = MiniFloat.from(-1f)
+  val One: MiniFloat = MiniFloat.from(1f)
+
+  val MaxValue: MiniFloat = Finite.max
+  val MinValue: MiniFloat = Finite.min
+  val MinPositiveValue: MiniFloat = Finite.minPositive
+
+  def allValues: List[MiniFloat] =
+    List(NegativeInfinity) ++
+      Finite.allValues :+
+      PositiveInfinity :+
+      NaN
+
+  def from(float: Float): MiniFloat =
+    float match {
+      case Float.PositiveInfinity => PositiveInfinity
+      case Float.NegativeInfinity => NegativeInfinity
+      case f if f.isNaN           => NaN
+      case _ =>
+        Finite
+          .from(float)
+          .getOrElse {
+            if (float > 0) PositiveInfinity else NegativeInfinity
+          }
+    }
+
+  def from(double: Double): MiniFloat = from(double.toFloat)
+  def from(int: Int): MiniFloat = from(int.toFloat)
+  def from(long: Long): MiniFloat = from(long.toFloat)
+
+  /**
+   * Note that since `MiniFloat` is used primarily for tests, this `Eq` instance defines `NaN` as equal to itself. This
+   * differs from the `Order` defined for `Float`.
+   */
+  implicit val catsLawsEqInstancesForMiniFloat: Order[MiniFloat] with Hash[MiniFloat] =
+    new Order[MiniFloat] with Hash[MiniFloat] {
+      override def compare(x: MiniFloat, y: MiniFloat): Int = Order[Float].compare(x.toFloat, y.toFloat)
+      override def hash(x: MiniFloat): Int = Hash[Float].hash(x.toFloat)
+    }
```

Comment:
Because we're using this class in tests it's really only useful if we define `Eq` such that `NaN == NaN`. I'm aware that this is slightly unintuitive but otherwise it becomes useless.

An alternative approach would be to provide this `Eq` in such a way that it needs to be manually imported when used. I'd be happy to make that change as needed.